timemachine/fe/free_energy.py

```diff
@@ -1298,7 +1257,13 @@ def sample_replica(xvb: CoordsVelBox, state_idx: StateIdx) -> Tuple[NDArray, NDA
                 # Set the step so that all windows have the movers be called the same number of times.
                 mover.set_step(current_step)
 
-            md_params_replica = replace(md_params, n_frames=1, n_eq_steps=0, seed=state_idx + current_frame)
+            md_params_replica = replace(
+                md_params,
+                n_frames=1,
+                # Run equilibration as part of the first frame
+                n_eq_steps=md_params.n_eq_steps if current_frame == 0 else 0,
```

Comment:
This does seem a bit confusing and tech-debt-y. Is there a way to avoid duplicated code while also avoiding the need to hack the MDParams like this?

Reply:
I find this to be less tech-debt than where we currently split out equilibration where it could deviate from our reference version of equilibration (at least I interpret the code to be in `sample` to be the definitive version). Could make this an if/else to make it more readable, but nothing else immediately jumps out.

Guess you could also run `sample_context_with_iter` on all of the frames upfront with `n_frames == 0`, but notably different to what this is doing.

Reply:
Agreed that duplication is the least desirable option. Would it make sense to instead pull the reference equilibration code out of `sample_with_iter` into a separate function, and call this function both from `sample_with_iter` and `run_sims_hrex`?

Reply:
We could do that, though I am not clear what this is resolving beyond readability.

We are already modifying the MDParams on each state, dependent on the current frame. And if we end up parallelizing this function we will have to parallelize two call sites rather than one, which seems like a more serious bit of tech debt. Though maybe should be less focused on the latter for this PR, but these changes were spawned by my parallelization attempts.

Reply:
No reason other than readability/subjective preference for "decoupling". Not blocking, feel free to disregard